tests/Unit/Domain/Creators/TimeDependence/Test_Composition.cpp

```diff
@@ -0,0 +1,249 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <random>
+
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.tpp"
+#include "Domain/CoordinateMaps/ProductMapsTimeDep.hpp"
+#include "Domain/CoordinateMaps/ProductMapsTimeDep.tpp"
+#include "Domain/CoordinateMaps/Translation.hpp"
+#include "Domain/Creators/TimeDependence/Composition.hpp"
+#include "Domain/Creators/TimeDependence/Composition.tpp"
+#include "Domain/Creators/TimeDependence/TimeDependence.hpp"
+#include "Domain/Creators/TimeDependence/UniformTranslation.hpp"
+#include "tests/Unit/TestCreation.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+#include "tests/Utilities/MakeWithRandomValues.hpp"
+
+namespace domain {
+namespace creators {
+namespace time_dependence {
+
+namespace {
+using Translation = domain::CoordMapsTimeDependent::Translation;
+
+template <typename T0, size_t MeshDim>
+void test_impl(
+    const gsl::not_null<T0> gen, const double initial_time,
+    const std::unique_ptr<
+        domain::creators::time_dependence::TimeDependence<MeshDim>>& time_dep,
+    const std::unique_ptr<
+        domain::creators::time_dependence::TimeDependence<MeshDim>>&
+        expected_time_dep,
+    const std::vector<std::string>& expected_f_of_t_names) noexcept {
+  // Test coordinate maps
+  UniformCustomDistribution<size_t> dist_size_t{1, 10};
+  const size_t num_blocks = dist_size_t(*gen);
+  CAPTURE(num_blocks);
+
+  const auto functions_of_time = time_dep->functions_of_time();
+  REQUIRE(functions_of_time.size() == expected_f_of_t_names.size());
+  for (const auto& f_of_t_name : expected_f_of_t_names) {
+    CHECK(functions_of_time.count(f_of_t_name) == 1);
+  }
+
+  const auto functions_of_time_for_expected =
+      expected_time_dep->functions_of_time();
+
+  // For a random point at a random time check that the values agree. This is to
+  // check that the internals were assigned the correct function of times.
+  UniformCustomDistribution<double> dist{-1.0, 1.0};
```

Comment:
much of this seems pretty similar to the test for the previous commit. Can it be combined into a shared test helper?